contracts/BoostPass.sol

```diff
@@ -61,6 +62,7 @@ contract BoostPass is Initializable, Ownable, ERC721 {
     /// @param data_ The data to decode for the claim
     function mint(bytes calldata signature_, bytes calldata data_) external payable {
         (address to_, address referrer_) = abi.decode(data_, (address, address));
+        if (to_ != msg.sender) revert ToAddressIsNotSender();
```

Comment:
I had to include this check because then someone else would be able to use a different `signature_` and `data_` that doesn't match them

aka if this check were not here, then I'd be able to reuse existing params amongst tons of different addresses

Reply:
but totally get the EOA/abstraction thing... open to making any changes here

Reply:
They can use different `signature_` and `data_` but the token still goes to `to_` not `msg.sender`

Reply:
Actually this is an excellent point